tests/php/ORM/DataObjectTest.php

```diff
@@ -1405,27 +1411,37 @@ public function testPopulateDefaults()
         );
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     */
     public function testValidateModelDefinitionsFailsWithArray()
     {
         Config::modify()->merge(DataObjectTest\Team::class, 'has_one', array('NotValid' => array('NoArraysAllowed')));
-        $this->setExpectedException(InvalidArgumentException::class);
         DataObject::getSchema()->hasOneComponent(DataObjectTest\Team::class, 'NotValid');
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     */
     public function testValidateModelDefinitionsFailsWithIntKey()
     {
         Config::modify()->set(DataObjectTest\Team::class, 'has_many', array(0 => DataObjectTest\Player::class));
-        $this->setExpectedException(InvalidArgumentException::class);
         DataObject::getSchema()->hasManyComponent(DataObjectTest\Team::class, 0);
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     */
     public function testValidateModelDefinitionsFailsWithIntValue()
     {
         Config::modify()->merge(DataObjectTest\Team::class, 'many_many', array('Players' => 12));
-        $this->setExpectedException(InvalidArgumentException::class);
         DataObject::getSchema()->manyManyComponent(DataObjectTest\Team::class, 'Players');
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage Controller is not a valid subclass of DataObject
```

Comment:
As mentioned elsewhere, for mixed tests it's advisable to PHP api to set expected exception prior to the throwing expression, rather than annotations.